sale_payment_sheet/models/sale_payment_sheet.py

```diff
@@ -0,0 +1,255 @@
+        index=True,
+        copy=False,
+        default=fields.Date.context_today,
+    )
+    state = fields.Selection(
+        [("open", "New"), ("confirm", "Validated")],
+        string="Status",
+        required=True,
+        readonly=True,
+        copy=False,
+        default="open",
+    )
+    currency_id = fields.Many2one(
+        "res.currency", compute="_compute_currency", string="Currency"
+    )
+    journal_id = fields.Many2one(
+        "account.journal",
+        string="Journal",
+        required=True,
+        states={"confirm": [("readonly", True)]},
+        default=lambda self: self._default_journal(),
+    )
+    commercial_journal_ids = fields.Many2many(related="user_id.commercial_journal_ids")
+    company_id = fields.Many2one(
+        "res.company",
+        related="journal_id.company_id",
+        string="Company",
+        store=True,
+        readonly=True,
+        default=lambda self: self.env.company,
+    )
+    line_ids = fields.One2many(
+        "sale.payment.sheet.line",
+        "sheet_id",
+        string="Sheet lines",
+        states={"confirm": [("readonly", True)]},
+        copy=True,
+    )
+    user_id = fields.Many2one(
+        "res.users",
+        string="Responsible",
+        required=False,
+        default=lambda self: self.env.user,
+    )
+    statement_id = fields.Many2one(
+        comodel_name="account.bank.statement", string="Bank statement"
+    )
+
+    @api.model
+    def _default_journal(self):
+        return self.env.user.commercial_journal_ids[:1]
+
+    @api.depends("journal_id")
+    def _compute_currency(self):
+        for sheet in self:
+            sheet.currency_id = (
+                sheet.journal_id.currency_id or sheet.company_id.currency_id
+            )
+
+    @api.depends("journal_id", "user_id", "date")
+    def _compute_name(self):
+        for sheet in self:
+            sheet.name = "{} - {} - {}".format(
+                sheet.date and sheet.date.strftime("%Y.%m.%d"),
+                sheet.journal_id.name,
+                sheet.user_id.name,
+            )
+
+    def unlink(self):
+        for sheet in self:
+            if sheet.state != "open":
+                raise UserError(
+                    _("You can not delete a sheet if has related journal items.")
+                )
+        return super().unlink()
+
+    def button_confirm_sheet(self):
+        sheets = self.filtered(lambda r: r.state == "open")
+        BankStatement = self.env["account.bank.statement"].sudo()
+        BankStatementLine = self.env["account.bank.statement.line"].sudo()
+        for sheet in sheets:
+            statement = BankStatement.create(
+                {
+                    "name": sheet.name,
+                    "date": sheet.date,
+                    "journal_id": sheet.journal_id.id,
+                    "user_id": sheet.user_id.id,
+                }
+            )
+            for line in sheet.line_ids:
```

Comment:
Why not using the create multi ?

Reply:
Because I am linking each statement line to the sheet line.